src/token/src/canister.rs

```diff
@@ -185,228 +80,20 @@ impl TokenCanister {
-        }
-    }
-
-    /********************** AUCTION ***********************/
-
-    /// Bid cycles for the next cycle auction.
-    ///
-    /// This method must be called with the cycles provided in the call. The amount of cycles cannot be
-    /// less than 1_000_000. The provided cycles are accepted by the canister, and the user bid is
-    /// saved for the next auction.
-    #[update]
-    pub fn bidCycles(&self, bidder: Principal) -> Result<u64, AuctionError> {
-        bid_cycles(self, bidder)
-    }
-
-    /// Current information about bids and auction.
-    #[query]
-    pub fn biddingInfo(&self) -> BiddingInfo {
-        bidding_info(self)
-    }
-
-    /// Starts the cycle auction.
-    ///
-    /// This method can be called only once in a [BiddingState.auction_period]. If the time elapsed
-    /// since the last auction is less than the set period, [AuctionError::TooEarly] will be returned.
-    ///
-    /// The auction will distribute the accumulated fees in proportion to the user cycle bids, and
-    /// then will update the fee ratio until the next auction.
-    #[update]
-    pub fn runAuction(&self) -> Result<AuctionInfo, AuctionError> {
-        run_auction(self)
-    }
-
-    /// Returns the information about a previously held auction.
-    #[query]
-    pub fn auctionInfo(&self, id: usize) -> Result<AuctionInfo, AuctionError> {
-        auction_info(self, id)
-    }
-
-    /// Returns the minimum cycles set for the canister.
-    ///
-    /// This value affects the fee ratio set by the auctions. The more cycles available in the canister
-    /// the less proportion of the fees will be transferred to the auction participants. If the amount
-    /// of cycles in the canister drops below this value, all the fees will be used for cycle auction.
-    #[query]
-    pub fn getMinCycles(&self) -> u64 {
-        self.state.borrow().stats.min_cycles
-    }
-
-    /// Sets the minimum cycles for the canister. For more information about this value, read [get_min_cycles].
-    ///
-    /// Only the owner is allowed to call this method.
-    #[update]
-    pub fn setMinCycles(&self, min_cycles: u64) -> Result<(), TxError> {
-        let caller = CheckedPrincipal::owner(&self.state.borrow_mut().stats)?;
-        self.update_stats(caller, CanisterUpdate::MinCycles(min_cycles));
-        Ok(())
+impl ISTokenCanister for TokenCanister {
+    fn state(&self) -> Rc<RefCell<CanisterState>> {
+        self.state.clone()
     }
 
-    /// Sets the minimum time between two consecutive auctions, in seconds.
-    ///
-    /// Only the owner is allowed to call this method.
-    #[update]
-    pub fn setAuctionPeriod(&self, period_sec: u64) -> Result<(), TxError> {
-        let caller = CheckedPrincipal::owner(&self.state.borrow_mut().stats)?;
-        // IC timestamp is in nanoseconds, thus multiplying
-        self.update_stats(caller, CanisterUpdate::AuctionPeriod(period_sec));
-        Ok(())
+    fn canister(&self) -> &TokenCanister {
```

Comment:
Why is this needed?

Reply:
I was looking for a method on how to use the canister for the transfer methods.

Reply:
Well, this is an `id` function, you can use `self` as is 🤔

Reply:
I don't understand this, what do you mean exactly ?